nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -45,25 +52,48 @@
     @Override
     public DocumentModel createCompoundDocument(CoreSession session, String parent, Blob archive) {
         String compoundDocName = FilenameUtils.removeExtension(archive.getFilename());
-        DocumentModel compoundDoc = session.createDocumentModel(parent, compoundDocName, COMPOUND_DOCTYPE);
-        compoundDoc.setPropertyValue("dc:title", compoundDocName);
-        compoundDoc = session.createDocument(compoundDoc);
-        createCompoundDocument(compoundDoc, archive);
-        return compoundDoc;
+        var parentDoc = session.getDocument(new PathRef(parent));
+        try (var zip = new ZipFile(archive.getFile())) {
+            Set<String> entries = zip.stream().map(ZipEntry::getName).collect(Collectors.toSet());
```

Comment:
Here I'm only collecting the names to avoid giving the blobs to the script. Since we said we wanted to avoid giving the main blob, I thought the same about the gather of all the entries?

Reply:
I mean, why not just doing:
```java
List<ZipEntry> entries = zip.stream()
           .filter(this::isAllowedEntry)
           .sorted(Comparator.comparing(ZipEntry::getName))
           .collect(Collectors.toList())
```
to store the whole list of entries first, to avoid re-streaming the ZIP file in `#createCompoundDocument`